File: CPP/No470.cc
/**
 * Created by Xiaozhong on 2020/9/28.
 * Copyright (c) 2020/9/28 Xiaozhong. All rights reserved.
 */

// The rand7() API is already defined for you.
// @return a random integer in the range 1 to 7

int rand7();

class Solution {
public:
    int rand10() {
        int n = (rand7() - 1) * 7 + rand7();
        while (n > 10) {
            n = (rand7() - 1) * 7 + rand7();
        }
        return n;
    }
};

File: Java/src/No470.java
/*
 * Copyright (c) 9/6/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No470 {

    static class SolBase {
        public int rand7() {
            return 0;
        }
    }

    static class Solution extends SolBase {
        public int rand10() {
            int ret = (rand7() - 1) * 7 + rand7();
            while (ret > 10) ret = (rand7() - 1) * 7 + rand7();
            return ret;
        }
    }
}
